### DISCUSSION TITLE
The pinia documentation states that plug-ins can implement action cancellation. I didn’t find any specific examples, and I didn’t find out how to implement it based on the existing API.

### DISCUSSION THREAD

**Discussion by bayun2** [vuejs/pinia#2560] [Help and Questions] [answered]
The pinia documentation states that plug-ins can implement action cancellation. I didn’t find any specific examples, and I didn’t find out how to implement it based on the existing API.

<img width="801" alt="image" src="https://github.com/vuejs/pinia/assets/3183696/d1305423-4830-4d2e-92f9-e5b6cc745ea9">

I hope to make an middleware for the action behavior. When the specified conditions are not met, the action will be canceled instead of adding this judgment separately for each action.

**Comment by posva** [comment, marked as answer]
I'm not sure if the Chinese docs are up to date in that list but you should find this [issue](https://github.com/vuejs/pinia/issues/995) helpful to implement what you want
